cmd/influxd/run/config.go

```diff
@@ -175,6 +175,9 @@ func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value) error {
 			// e.g. GRAPHITE_0
 			if f.Kind() == reflect.Slice || f.Kind() == reflect.Array {
 				for i := 0; i < f.Len(); i++ {
+					if err := c.applyEnvOverrides(key, f.Index(i)); err != nil {
```

Comment:
The unit test doesn't work with the current code because inputs which allow multiple listeners don't get matched to environment variables like `INFLUXDB_UDP_BIND_ADDRESS`, only `INFLUXDB_UDP_[i]_BIND_ADDRESS`. This change allows both to be used to override the bind address for a section and only applies to sections which allow multiple instances, which are all input sections (UDP, graphite, etc).

Not sure how this would match a `INFLUXDB_0_BIND_ADDRESS` env var to all bind addresses. Maybe I'm missing something.